src/arviz_stats/base/dataarray.py

```diff
@@ -193,5 +193,30 @@ def kde(self, da, dims=None, circular=False, grid_len=512, **kwargs):
         out = concat((grid, pdf), dim=plot_axis)
         return out.assign_coords({"bw" if da.name is None else f"bw_{da.name}": bw})
 
+    def thin(self, da, factor="auto", dims=None):
```

Comment:
It is ok to have this logic here, or it should be inside src/arviz_stats/base/array.py (see commented code). Maybe ok because we are just slicing the DataArray so no need to have ufuncs and all that???

Reply:
I think here it is preferable to implement the function here. If eventually there are libraries actually using the array only backend we can implement an independent version there and not use it here. The reason why is if we do this via the array backend, the dimension being modified (generally `draw` here) will have to be moved to the last position. Dimension order should be taken for granted in xarray, but it is also nice to not change the order unless necessary, and here it isn't